lib/OpenQA/Schema/Result/Comments.pm

```diff
@@ -169,4 +170,29 @@ sub extended_hash ($self) {
     };
 }
 
+sub handle_special_contents ($self, $c = undef) {
+    $self->_insert_bugs;
+    $self->_control_job_result($c);
+}
+
+sub _control_job_result ($self, $c) {
+    return undef unless my ($new_result, $description) = $self->force_result;
+    return undef unless $new_result;
+    die "Invalid result '$new_result' for force_result\n"
+      unless grep { $_ eq $new_result } OpenQA::Jobs::Constants::RESULTS;
+    die "force_result labels only allowed for operators\n" unless !$c || $c->is_operator;
```

Comment:
I usually avoid `unless` with more than one condition, especially as the first one is a negation. How about:
```suggestion
    die "force_result labels only allowed for operators\n" if $c && !$c->is_operator;
```